src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -459,6 +459,24 @@ private function bail( string $message ) {
 		throw $this->unsupported_exception;
 	}
 
+	/**
+	 * Sets the document compatibility mode (quirks or no-quirks) based on a DOCTYPE declaration.
+	 *
+	 * @see https://html.spec.whatwg.org/multipage/parsing.html#parser-cannot-change-the-mode-flag
+	 *
+	 * @since 6.7.0
+	 */
+	private function update_document_mode_from_doctype(): void {
+		$doctype = $this->get_doctype_info();
+		if ( null === $doctype ) {
+			return;
+		}
+
+		if ( 'quirks' === $doctype->get_compatibility_mode() ) {
```

Comment:
this reads a bit awkward to me. also one of the reasons I find direct property access helpful. the DOCTYPE token doesn't specify a document's compatibility mode - it can only suggest it when none is already provided or forced, right?

so we might say "this token indicates quirks mode" or "this token indicates no-quirks mode"

```
if ( 'quirks' === $doctype->indicated_compatability_mode ) { }
```

but on this note do we want to constrain this function to only operate on a DOCTYPE token? I would imagine this could lead to people needing to try and push their own doctype declaration into existing HTML, when they could instead set this instead

```
$processor->set_compatability_mode( WP_HTML_Tag_Processor::QUIRKS_MODE );
$processor->set_compatability_mode( WP_HTML_Tag_Processor::NO_QUIRKS_MODE );
```

Although this does in a sense expose an opportunity for a foot-gun, for someone to change the compatibility mode randomly, it doesn't force new rules onto the Tag Processor that I don't think fully exist there. we're not going to have a token in the fragment case you mention, so maybe we analyze the parsed DOCTYPE in the HTML Processor and then change modes in the Tag Processor via method call.

we can even `_doing_it_wrong()` if setting the compatibility mode after having already set it, which will send an error in debug modes but won't take down the site.